### ISSUE TITLE
Ping from a webserver

### ISSUE THREAD

**Issue by FrankC2000** [BlakeFoster/Arduino-Ping#12]
Nice work here! Thanks!

My purpose is to build a little webserver in a box. I will put a led outside to indicate the whether network is on (connected). So I ping the gateway eache 5 secs to set the led.

The problem is, as soon as the server starts to receive HTTP requests, it doesn't ping the gateway anymore, I mean does not receive SUCCESS status. Could be a socket issue?

Any thoughts? Thanks in advance.

BTW, a changed pingSocket from 0 to 3 but it didn't work.

Keep up the good work!

**Comment by BlakeFoster** [comment]
Yes, it's probably a socket conflict. I'm not sure what libraries you're 
using for the HTTP server, but I would look there to see how it's 
picking sockets. It's probably paving over the raw socket used for ICMP.
-Blake

FrankC2000 wrote:

**Comment by FrankC2000** [comment]
Thank you for your quick reply! Almost a ping ;)

I extended the Ethernet/Webserver example that comes with the IDE. The library is Ethernet.h. I have no idea what to change there.

- Frank

**Comment by BlakeFoster** [comment]
If I remember correctly, the Ethernet library automatically picks a 
socket that is not in use. I would suggest initializing the ICMP socket 
before doing anything else with the Ethernet, if you're not already. In 
theory, the Ethernet library should detect that you're using that socket 
and skip it (probably--I don't actually remember how it detects whether 
a socket is being used). There are only 4 sockets, so your chances of 
getting a conflict if you don't take steps to avoid one are high.
-Blake

FrankC2000 wrote:

**Comment by FrankC2000** [comment]
What do you mean by initializing the ICMP socket? The very first call to the ICMPEchoReply.ping method? 

I call Ethernet.begin before the first call to the ICMPEchoReply.ping.
